src/timeline/index.ts

```diff
@@ -0,0 +1,114 @@
+import createOpen from './before/create';
+import * as log from '../log';
+
+
+export type condition = (events:TimelineEvent[], event?:TimelineEvent) => boolean;
+
+export const getTime = 'now' in performance ? () => {
+    return performance.now()
+} : () => {
+    return (new Date()).getTime()
+};
+
+const beforeTest:condition[] = [createOpen];
+const afterTest = [];
+
+class Timeline {
```

Comment:
This class should have a very detailed comment as it provides the basis for the popup blocking functionality.